### DISCUSSION TITLE
Trocar o Sublime por VSCode

### DISCUSSION THREAD

**Discussion by robsonbittencourt** [frontendbr/forum#321] [Perguntas e Respostas]
[Publiquei um post a pouco](http://rbittencourt.com/vscode-meu-novo-editor-de-textos/) e gostaria de trazer a discussão para cá.

Durante muito tempo utilizei o Sublime Text como editor. Várias pessoas no meu trabalho estavam utilizando o VSCode e eu seguia abraçado no Sublime. 

Esta semana decidi fazer um teste e agora não quero mais voltar :)

Os principais pontos que fizeram eu tomar esta decisão:
- Free e open source
- Debugger para Node
- Integração muito boa com o Git

Desvantagens:
- O start é bem mais lento que o do Sublime. 

O que vocês acham sobre está mudança? Mais alguém passou por ela recentemente? Estão curtindo, ou tendo problemas?

**Comment by YuriBrunetto** [comment]
Nunca tinha ouvido falar do VSCode, mas me pareceu bem interessante. Vou baixá-lo e testar! 👍 
(vale mencionar que atualmente estou usando o Atom, heheh)

**Comment by felipefialho** [comment]
@robsonbittencourt Cara, já tentei, mas não consigo abrir mão do Sublime. Tem tudo que preciso e é rápido, haha.

**Comment by danguilherme** [comment]
Eu usava Sublime, comecei a usar o Atom e depois, o VSCode. Valeu a pena! Não lembro do tempo de startup do Sublime, mas o VSCode abre muito mais rápido que o Atom.

**Comment by felipefialho** [comment]
Não gostei do Atom, tem potencial, mas não conseguiu me deixar satisfeito ainda. 

Em tempos @robsonbittencourt, gostei do seu review, deu até vontade de testar.

**Comment by robsonbittencourt** [comment]
Não testei o Atom ainda, mas vi pessoas reclamando que ele é pesado.

@LFeh o que você tem no Sublime que não encontrou no VSCode?

**Comment by felipefialho** [comment]
Editei a última resposta, na verdade não testei o VSCode ainda 😄

**Comment by robsonbittencourt** [comment]
Ahh ta :smile: 

Imagino que se você utiliza muitos plugins no Sublime, não irá encontrar todos no VSCode, e isso pode ser um problema para migrar mesmo.

**Comment by RuanAragao** [comment]
Muito bom o post @robsonbittencourt, fiquei com vontade de testar, e irei fazer isso. Gostava bastante do Sublime Text, mas acabei abandonando ele, fiquei utilizando Atom, mas começou a me dar muita dor de cabeça quando intensifiquei meu fluxo de trabalho. Acabei por migrar de vez para o Vim, mais uma vez. Atualmente estou em paz, mas irei testar o VSCode. :)

**Comment by casssiahaluu** [comment]
Boa noite
Já usei Os três citados. Como uso Ubuntu com uma máquina mais velhinha, tenho preferência por editores de texto à IDEs. No Ubuntu, acho o Sublime tão pesado quanto o PyCharm, por exemplo, porém, o Atom com packs demais também fica pesadinho. O VScode é o mais leve de todos. A desvantagem do VScode, como já disseram, é a falta de alguns packs, porém, muita coisa que no Sublime e no Atom você procuraria packs pra ter, nele você já tem por default. Eu uso atualmente o VScode para desenvolver em EF core e Atom para as demais coisas, já que coisas que envolvam c# o produto da Microsoft se sai muito melhor, mas o dia em que houver packs melhores para VScode, migro por completo pra ele.

**Comment by akfzambrana** [comment]
No meu notebook ainda uso sublime, amo ele e não trocaria por opção propria, ele faz tudo que preciso e a busca de arquivos ou de palavras, é simplesmente insuperável.
Na empresa **fomos obrigados** a trocar de editor por questão de politica de licenças. Como o sublime não é freeware tivemos que desinstalar ele de todas as máquinas e optar por uma opção free. 
Dessas opções free, **o único que realmente vale a pena é o VSCode**, os outros que testei (Atom e Brackets) são lentos d+ e bugados d+.
Honestamente, estou utilizando o VSCode faz aproximadamente um mês e hoje só sinto falta do sublime na hora de procurar por algum arquivo 😢 então acredito que vale a pena a mundaça sim.

**Comment by vitor-mendrone-deel** [comment]
Reza a lenda que o VSCode nada mais é do que um fork do Atom. A lenda procede? 

Tentei utilizar o Atom, fiquei com ele por volta de 2 meses. É um editor incrível, organizado e bonito, mas a performance é nojenta em comparação com o Sublime e o consumo de memória é insano, Manter abertos o Slack, Atom, Chrome e Avocode (um programa pra extração de assets de PSDs) fazia um Intel Core i7 de quarta geração com 16 gigas de ram funcionar como uma lesma. 

Voltei pro Sublime e o problema acabou.

**Comment by douglas-castilho** [comment]
não sei se a lenda procede @Mendrone, mas se for eles deram uma boa turbinada, estou testando o VSCode e em questão de desempenho está sendo beeeeemmm melhor que o atom ou brakets

**Comment by robsonbittencourt** [comment]
@akfzambrana não notei diferença na busca por arquivos comparando com o Sublime. 

Basta apertar ctrl + p e sair digitando. Inclusive aquele lance (não sei o nome) de você digitar a primeira letra de cada palavra do arquivo funciona no VSCode também.

**Comment by akfzambrana** [comment]
Aqui trabalhamos com uma VM linux, pois o ambiente da empresa default é windows (n podemos mudar 😢), seguindo este ambiente, com o sublime instalado na maquina windows a busca por qualquer arquivo dentro da VM é instantâneo. Ja o VSCode demora entre 3 a 5 segundos pra achar o arquivo, sabendo o nome ou não. Claro que estou falando de arquivos que não foram abertos ainda, os que já foram abertos ele acha instantaneamente.

**Comment by napalmdev** [comment]
Eu migrei do Sublime do VSCode a um tempinho atrás, eu tinha tentado utilizar ele ao um tempo atrás quando lançou e não curti, era meio pesadão mesmo. 
Porém, um belo dia após planejar a sprint a galera ficou trocando ideia e surgiu um papo sobre o VSCode, ai todo mundo começou a testar e fomos falando sobre os plugins q precisávamos e no fim da reunião, todo mundo tava utilizando o VSCode, e desde então não tive nem interesse em voltar ao Sublime mais.

O VSCode tem muitos recursos matadores e alguns plugins que são fora de série, por exemplo a integração com Git dele é sensacional a forma como ele faz o Diff dos arquivos, além de atalhos pra fazer add, commit e checkout de arquivos, é foda d+, outra coisa linda é a integração dele com tarefa dos task runners, vc consegue rodar tarefas direto dentro dele, ele tem um terminal integrado tbm caso precise fazer alguma coisa, o Debugger dele tbm é sensacional, hahah to vendido pelo VSCode.

E ainda tem esta extensão aqui que permite q vc salve todas as suas configurações, snippets, extensões e afins em uma Gist secreta direto no Github e quando migrar de máquina é só utilizar a extensão pra sincronizar e configurar tudo automaticamente ________   é lindo d+
https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync

E quanto a performance, na primeira vez q eu tinha utilizado ele era bem lentão mesmo, tipo o Atom, porém nestas ultimas versões eu sinto ele tão leve de se utilizar quanto o Sublime, sérião mesmo, testem e depois deêm seus vereditos kkkkkkk

**Comment by robsonbittencourt** [comment]
Pior que também fiquei vendido para o VSCode @napalmdev. 

Sobre a performance aqui leva uns 6 segundos para dar o start e minha máquina é boazinha. Uma coisa que estava pensando sobre isso, é que estou usando o Insiders (que é a versão do último build) vou ver se faço um teste com a corrente.

**Comment by ghost** [comment]
Gostei bastante do vscode, realmente ele é muito bom, até melhor que o Atom pois:
- Possui integração nativa com o Git
- Tem botões laterais que são muito práticos 
- Reinicia automaticamente após instalar extensões

Recomendo! Fazia tempo que queria testar, valeu pelo incentivo @robsonbittencourt  =)

**Comment by rodrigocipriani** [comment]
Boa noite pessoal, nos meus testes percebi o seguinte. 
*resumindo bastante.
- Atom bem bom, mas tem travado um pouco.
- VSCode achei nem leve e prático, mas tem travado quando o projeto tem muitos arquivos. (Vulgo node_modules)
- Sublime não testei o suficiente para opinar. Mas ele é pago não? 
- Outro que é pago é o WebStorm. Esse sim, esse com toda certeza é o melhor que já utilizei. 
  Alguém mais já utilizou?

**Comment by ghost** [comment]
@rodrigocipriani sim, o sublime é pago mas pode ser usado gratuitamente. volta e meia aparece uns alerts para registrar ele, mas nada que o inutilize.

também gosto do atom, em que ambiente você tem usado ele? tanto o vscode quanto o atom usam o electrom como base.

curto bastante as ides da jetbrains, o problema é que elas são muito pesadas (java) e a maioria delas é paga. quando era mais noob usava bastante o phpstorm, mas acabei trocando ele pelo atom.

**Comment by rodrigocipriani** [comment]
Eu uso com node, Angular, Angular 2, React, React   Native, Express, Sequelize, etc. Ambiente Javascript mesmo.
Quanto a ser pesado o WebStorm, para mim ele é o único que não trava nunca.
Outra coisa, geralmente nem precisa instalar plugin, ele já vem completão, pra qqr ambiente. É muito top.

**Comment by ghost** [comment]
@rodrigocipriani desculpa, quis dizer plataforma. Parece que no Mac o Atom costuma travar bastante.

**Meu veredicto após testar o vscode:**
Depois de ter travado e reiniciado o pc duas vezes desinstalei ele e voltei pro bom e velho Atom.

**Comment by rodrigocipriani** [comment]
@HackerGaucho Eu uso Windows. No Linux o cenário muda, parece que trava muito menos.

**Comment by ghost** [comment]
Sim, no Ubuntu aqui é tranquilo, bem raro travar e quanto trava é só matar o processo que depois ele volta suave.

**Comment by ghost** [comment]
Instalar o vscode me fez voltar aos tempos do Windows quando o computador travava e tinha que reiniciar. Só a Microsoft pra conseguir repetir essa façanha no Ubuntu! tsc tsc

**Comment by raphaguasta** [comment]
Fala Pessoal!

Trabalho usando Mac, e meu editor preferido era o Sublime Text há tempos.

Eu tinha um certo preconceito com o VSCode, pois já utilizei o Visual Studio que era muitoo lento e traumatizou, rs.. Mas dei uma chance ao VSCode, e agora não volto mais para o sublime!

Variedade de plugins, atualizações rápidas (tanto do software quanto dos plugins), terminal integrado (me ajuda muuito), autocomplete e IntelliSense funcionam bem... Essas coisas me fizeram adotá-lo de vez!
A única desvantagem realmente é a inicialização lenta, como citou o @robsonbittencourt .

**Comment by rogeralbinoi** [comment]
Cara, eu estou usando o VSCode agora, e estou curtindo muito.

**Comment by rodrigocipriani** [comment]
Eu também curto bastante o vscode. No meu caso ele tem travado algumas
vezes.
Dos gratuitos, ele ainda é minha melhor opção. Mas incluindo pagos,
Webstorm é muito melhor.

**Comment by rodrigocipriani** [comment]
Como eu deleto a mensagem anterior. Enviei a resposta pelo e-mail, mas não excluí os dados anteriores do email.

**Comment by alexsandro-xpt** [comment]
Eu usava o Sublime, mas migrei para o VSCode já tem um bom tempo, altamente recomendado  o VSCode.

**Comment by vitor-mendrone-deel** [comment]
@rodrigocipriani Removi os dados adicionais da sua resposta ok? Deixei só o texto acima dos dados do e-mail.

**Comment by rodrigocipriani** [comment]
Ótimo. Obrigado.
Se quiser, fique a vontade para apagar essa conversa paralela para não atrapalhar o assunto principal.
Abraço.

**Comment by dsoldera** [comment]
Testei aqui e achei bem interessante o VSCode. Só fiquei sentinfo falta de salvar o Projeto/Workspace e pode alternar entre eles, como no Sublime. Alguem acho algo a respeito disso?

**Comment by robsonbittencourt** [comment]
@dsoldera é só instalar esta extensão:

https://github.com/alefragnani/vscode-project-manager

**Comment by dsoldera** [comment]
Obrigada @robsonbittencourt

**Comment by allexon** [comment]
Olá, alguém sabe como tirar estas malditas bordas em marcação de tags e fechamentos de código!!! QUE BORDAS IRRITANTES, alguém me ajude por favor! veja abaixo o que eu estou falando!!!!!
![bordas](https://cloud.githubusercontent.com/assets/1206288/23332623/8debee08-fb5b-11e6-9475-0edef8228b97.jpg)

**Comment by calaca** [comment]
@allexon deve ter como remover essas bordas nas [configurações do editor](https://code.visualstudio.com/Docs/customization/userandworkspace).

**Comment by allexon** [comment]
Posso falar em português, Então Lorena eu Já tentei um monte de coisa,  mas não consigo achar onde eu tiro estas bordas, estou usando win7 32 com a ver vscode 1.9.1, estas bordas ficam irritantes, cria uma poluição visual e por incrível que pareça atrapalho um pouco a concentração, são irritantes!

**Comment by fabiorocha** [comment]
**Disclaimer: nem eu nem meu empregador nos responsabilizamos por quaisquer danos causados. Tente sob sua própria responsabilidade.**

Fala @allexon, esse é um cenário não suportado pelo VS Code. Então você não vai conseguir configurar o editor pra fazer isso através das settings. Se você quiser realmente se livrar daquelas bordas (bracket-matching) você vai ter que fazer o seguinte:

1. Feche seu VS Code.
2. Vá até o diretório onde seu VS Code está instalado. Por exemplo: "C:\Program Files (x86)\Microsoft VS Code\"
3. Procure pelo arquivo chamado: workbench.main.css
4. Abra o arquivo em outro editor, por exemplo: Notepad ou Sublime.
5. Procure por: 

`.monaco-editor .bracket-match{ box-sizing:border-box;background-color:rgba(0,100,0,.1) }`

6. Troque para:

`.monaco-editor .bracket-match{ box-sizing:border-box;background-color:rgba(0,100,0,.1);border:0 !important; }`

7. Salve o arquivo e abra o VS Code. Agora ele deve estar como você gostaria. Ele vai dar uma aviso que sua instalação está corrompida e você pode clicar no botão "Never show again", pra não ver mais essa notificação.

E é isso. Talvez você não tenha permissão pra editar esse arquivo porque geralmente arquivos que vivem abaixo de "C:\Program Files (x86)\" são protegidos contra write. Se você for admin na máquina você pode se dar acesso a esse arquivo e modificá-lo.

Claro que quando você fizer update do VS Code provavelmente isso vai ser sobrescrito, então eu recomendo que além de fazer isso, você abra uma issue em https://github.com/microsoft/vscode/issues.

Abs.

**Comment by allexon** [comment]
OBRIGADO! OBRIGADO!!!, do primeiro jeito que você não deu certo, ele até
abriu o vscode, mas não conseguia abrir nada!!! Então o que eu fiz foi o
seguinte, mandei localizar : bracket-match, e  onde tinha border: eu
informei none! Assinm funcionou sem as malditas bordas! MUITO
OBRIGADO!!!!!!!

bracket-match

Em 25 de fevereiro de 2017 15:53, Fabio Rocha <notifications@github.com>
escreveu:

> *Disclaimer: nem eu nem meu empregador nos responsabilizamos por quaisquer
> danos causados. Tente sob sua própria responsabilidade.*
>
> Fala @allexon <https://github.com/allexon>, esse é um cenário não
> suportado pelo VS Code. Então você não vai conseguir configurar o editor
> pra fazer isso. Se você quiser realmente se livrar daquelas bordas
> (bracket-matching) você vai ter que fazer o seguinte:
>
>    1. Feche seu VS Code.
>    2. Vá até o diretório onde seu VS Code está instalado. Por exemplo:
>    "C:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\
>    workbench\electron-browser"
>    3. Procure pelo arquivo chamado: workbench.main.css
>    4. Abra o arquivo em outro editor, por exemplo: Notepad ou Sublime.
>    5. Procure por:
>
> .monaco-editor .bracket-match{ box-sizing:border-box;
> background-color:rgba(0,100,0,.1) }
>
>    1. Troque para:
>
> .monaco-editor .bracket-match{box-shadow: none !important}
>
>    1. Salve o arquivo e abra o VS Code. Agora ele deve estar como você
>    gostaria. Ele vai dar uma aviso que sua instalação está corrompida e você
>    pode clicar no botão "Never show again", pra não ver mais essa notificação.
>
> E é isso. Talvez você não tenha permissão pra editar esse arquivo porque
> geralmente arquivos que vivem abaixo de "C:\Program Files (x86)" são
> protegidos contra write. Se você for admin na máquina você pode se dar
> acesso a esse arquivo e modificá-lo.
>
> Abs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/frontendbr/forum/issues/321#issuecomment-282504191>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABJoEODoVfMCeseefURwEaebRuGI7PXHks5rgHihgaJpZM4KStWK>
> .
>



-- 
Desde já Grato,
Alexon da Silva Moreira
Serra - ES

**Comment by fabiorocha** [comment]
@allexon Isso. Copiei a parte errada quando formulei a resposta mas logo corrigi. Aqui no GitHub tá atualizado com a resposta correta. De qualquer forma fico feliz que você resolveu seu problema. 

Abs
